### ISSUE TITLE
vector<string> does not work with new abi gen

### ISSUE THREAD

**Issue by lonkly** [EOSIO/eosio.cdt#109]
I am trying to embed a struct into another one, which is eosio::table.
```
    struct st_question {
        string title;
        vector<string> options;
        string type;
    };
```
vector<string> would generate `"type": "string>"` and nodeos can't parse that, giving `Error 3015004: The type defined in the ABI is invalid`

**Comment by lonkly** [comment]
I used` typedef std::vector<std::string> strings;`  and manually added 
```
    "types": [
        {
            "new_type_name": "uuid",
            "type": "int32"
        },
        {
            "new_type_name" : "strings",
            "type": "string[]"

        }
    ],
```
to .abi file. Then it magically worked, transaction passed. So guys, how do I properly declare a vector of strings with new CDT?

**Comment by oranke** [comment]
I have similar issue too. 

```cpp
        // @abi table tries i64
        struct [[eosio::table("tries")]] tries_info {
            int64_t         round;
            int64_t         seedValue; 

            vector< uint8_t > resultNumber;

            int64_t primary_key() const { return round; } 
            uint64_t get_descOrder() const { return -round; }
        };
        typedef eosio::multi_index<N(tries), tries_info
           , indexed_by<N(desc), const_mem_fun<tries_info, uint64_t, &tries_info::get_descOrder>>
        > tries;
```

with `eosiocpp`

```javascript
{
      "name": "tries_info",
      "base": "",
      "fields": [{
          "name": "round",
          "type": "int64"
        },{
          "name": "seedValue",
          "type": "int64"
        },{
          "name": "resultNumber",
          "type": "uint8[]"
        }
      ]
    }
```

but `eosio-cpp`

```javascript
        {
            "name": "tries_info",
            "base": "",
            "fields": [
                {
                    "name": "round",
                    "type": "int64"
                },
                {
                    "name": "seedValue",
                    "type": "int64"
                },
                {
                    "name": "resultNumber",
                    "type": "vector<uint8_t>" // !!!HERE
                }
            ]
        },
```
When I called action which ref's this info, Error printed like this.

```
Error 3015004: The type defined in the ABI is invalid
```

So I modified that type to "uint8_t[]" and It works well.

**Comment by aaaking** [comment]
I also encountered this situation, how to resolve it?  help me!!!

**Comment by cppfuns** [comment]
I also encountered this situation, how to resolve it? help me!!!

**Comment by lonkly** [comment]
come on guys, just edit .abi file and replace all vector types looking wrong (like string> or vector<uint8_t>) with string[] and uint8_t[]

**Comment by larryk85** [comment]
This has been resolved as of v1.3.0
